src/types.rs

```diff
@@ -65,7 +65,7 @@ impl PartialEq<MappingType> for AstarteType {
             };
         }
 
-        if other == &MappingType::LongInteger {
+        if other == &MappingType::LongInteger || other == &MappingType::Double {
```

Comment:
This fixes the problem, but there is a resulting behaviour that I would like to highlight.

For example. When receiving the data: `42` that is mapped in the interface as a `double` the `poll` function will return the value `AstarteType::Integer(42)`. 
The same problem was already present for the `longinteger` mapping. 
For example. When receiving the data `12` that is mapped in the interface as a `longinteger` the `poll` function will return the value `AstarteType::Integer(12)`.

For an interface of the type `double` the user should expect the `poll` function to produce an `AstarteType::Integer` or `AstarteType::Double`.

Reply:
Right, but in the case we have two options:
1. Because Astarte send the smallest type, also the SDK adapts to it.
2. The Astarte Device SDK should produce the AstarteType in according with the mapping of Interface, also because Rust doesn't allow to receive a different type.

Reply:
The second option could result in an easier to use API for a potential user.
With the current implementation each user needs an additional `match` block when receiving an endpoint that they expect to be a `Double` and then cast to an `f64`/`f32` separately for `Integer`/`Double`.
With option 2 they would just cast to `f64`/`f32` .

A third option might be to implement the `TryFrom` trait from `AstarteType::Integer` to `f32`/`f64`.

To be honest, regarding the first option I lack the knowledge to understand if that could be an ok behaviour or not. Not sure how the other SDK handle this, and how much it's a language related issue.